src/Halogen/Svg/Indexed.purs

```diff
@@ -26,94 +49,206 @@ type GlobalEventAttributes r =
   , onMouseOver :: MouseEvent
   , onMouseUp :: MouseEvent
   , onWheel :: WheelEvent
-  | r)
+  | r
+  )
 
--- These can also be done with CSS
-type PresentationAttributes r = (stroke :: String, fill :: String | r)
+type GlobalAttributes r = GlobalEventAttributes (CoreAttributes r)
 
-type MarkerAttributes r =
-  ( markerStart :: String
-  , markerMid :: String
-  , markerEnd :: String
+-- Presentation attributes, grouped by applicability (see table above) ---------
+type StrokeAttributes r =
+  ( stroke :: String
+  , strokeDasharray :: String
+  , strokeDashoffset :: Number
+  , strokeOpacity :: Number
+  , strokeWidth :: Number
   | r
   )
 
-type GlobalAttributes r = (PresentationAttributes (GlobalEventAttributes (StyleAttributes (CoreAttributes r))))
+type StokeEndAttributes r =
+  ( strokeLinecap :: String
+  | r
+  )
 
-type SVGsvg = GlobalAttributes
-  ( width :: Number
-  , height :: Number
-  , viewBox :: String
-  , preserveAspectRatio :: String
+type StrokeJoinAttributes r =
+  ( strokeLinejoin :: String
+  , strokeMiterlimit :: String
+  | r
   )
 
-type SVGcircle = GlobalAttributes
-  ( cx :: Number
-  , cy :: Number
-  , r :: Number
-  , transform :: String
+type FillAttributes r =
+  ( fill :: String
+  , fillOpacity :: Number
+  | r
   )
 
-type SVGellipse = GlobalAttributes
-  ( cx :: Number
-  , cy :: Number
-  , rx :: Number
-  , ry :: Number
-  , transform :: String
+type MarkerAttributes r =
+  ( markerStart :: String
+  , markerMid :: String
+  , markerEnd :: String
+  | r
   )
 
-type SVGrect = GlobalAttributes
-  ( x :: Number
-  , y :: Number
-  , rx :: Number
-  , ry :: Number
-  , width :: Number
-  , height :: Number
-  , transform :: String
+type FontAttributes r =
+  ( fontFamily :: String
+  , fontSize :: String
+  , fontSizeAdjust :: Number
+  , fontStretch :: String
+  , fontStyle :: String
+  , fontVariant :: String
+  , fontWeight :: String
+  | r
   )
 
-type SVGg = GlobalAttributes
-  ( transform :: String )
+type AllPresentationAttributes r
+  = StrokeAttributes
+      ( StrokeJoinAttributes
+          ( StokeEndAttributes
+              ( FillAttributes
```

Comment:
Could you replace all of this nesting with [`+`](https://pursuit.purescript.org/packages/purescript-typelevel-prelude/6.0.0/docs/Type.Row#t:type%20(+)) from `purescript-typelevel-prelude`? It'll require adding that dependency, but I think this will be more readable as a result.

Reply:
Thanks for the pointer to `purescript-typelevel-prelude`. I think the code is significantly cleaner as a result.